File: backend/databasehandler/src/main/java/com/pixtranslator/backend/databasehandler/repository/EnglishWordRepository.java
package com.pixtranslator.backend.databasehandler.repository;

import com.pixtranslator.backend.databasehandler.model.EnglishWord;
import com.pixtranslator.backend.databasehandler.model.GermanWord;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface EnglishWordRepository extends CrudRepository<EnglishWord, String> {
  List<EnglishWord> findAllByGermanword(GermanWord germanword);
}

File: backend/databasehandler/src/main/kotlin/com/pixtranslator/backend/databasehandler/repository/EnglishWordRepository.kt
package com.pixtranslator.backend.databasehandler.repository

import com.pixtranslator.backend.databasehandler.model.EnglishWord
import com.pixtranslator.backend.databasehandler.model.GermanWord
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface EnglishWordRepository : CrudRepository<EnglishWord, String> {
  fun findAllByGermanword(germanword: GermanWord?): List<EnglishWord>
}